src/packagedcode/__init__.py

```diff
@@ -253,3 +253,12 @@ def get_package_handler(package_data):
         raise UnknownPackageDatasource(package_data)
     return ppc
 
+
+PACKAGE_DATA_CLASS_WITH_CUSTOM_LICENSE_DETECTION = [
+    maven.MavenPackageData
+]
+
+PACKAGE_DATA_CLASS_BY_DATASOURCE_ID = {
```

Comment:
Create the data at once statically rather that doing this work on each import
```suggestion
PACKAGE_DATA_CLASS_BY_DATASOURCE_ID = {maven.MavenPackageData.datasource_id: maven.MavenPackageData}
```

Reply:
This would be better, doing this instead, thanks!